actions/lib/action.py

```diff
@@ -104,3 +109,31 @@ def _run_client_method(self, method, method_kwargs, format_func, format_kwargs=N
         result = method(**method_kwargs)
         result = format_func(result, **format_kwargs or {})
         return result
+
+    def _manipulate_rule(self, name, pack, enabled):
+        rule_name = '{}.{}'.format(pack, name)
+        failure_reason = None
+        rule = None
+        try:
+            rule = self.client.rules.get_by_name(name=name, pack=pack)
+            if not rule:
+                failure_reason = 'rule not found'
+        except Exception as exc:
+            failure_reason = exc
```

Comment:
why not just return here?

Reply:
To return a similarly formatted string of the error? Otherwise, it'd be two pretty much duplicate returns.

Reply:
What i'm suggesting is doing something like:
```
        try:
            rule = self.client.rules.get_by_name(name=name, pack=pack)
            if not rule:
                failure_reason = 'rule not found'
        except Exception as exc:
            return 'Could not get rule {}: {}'.format(rule_name, failure_reason)
```

Reply:
What if `self.client.rules.get_by_name` returns None? It won't be an exception and I'd like to stop execution and return.
And I think it's quite possible:
```py
    @add_auth_token_to_kwargs_from_env
    def get_by_name(self, name, **kwargs):
        instances = self.query(name=name, **kwargs)
        if not instances:
            return None
        else:
            if len(instances) > 1:
                raise Exception('More than one %s named "%s" are found.' %
                                (self.resource.__name__.lower(), name))
            return instances[0]
```

Reply:
Reached out to you on Slack to discuss this before I push my changes.